### ISSUE TITLE
[Customer]PowerKVM 2.1.1.4-70 can NOT boot from the right entry in Petitboot in xCAT 2.9.0

### ISSUE THREAD

**Issue by lxguang** [xcat2/xcat-core#266]
Hi, I am Liu XiaoGuang at IBM in CSTL. Just sitting beside xCAT team.
I am using the xCAT 2.9.0 shipped with PCM 4.2.1 to provision a Power8 8247-22L node with PowerKVM 2.1.1.4-70.  
When the PowerKVM OS installation is done, the installer will boot to finish the remaining installation.   
Yet, because the node was installed with Ubuntu 14.04.2 LE previously, there are multiple boot entries in the Petitboot menu.
The system will not go into PowerKVM. Instead it goes into Ubuntu. So there is an error.

**Comment by lxguang** [comment]
See the figure captured 
![2015-10-15_144140](https://cloud.githubusercontent.com/assets/15136852/10506506/3a35bc9c-734b-11e5-8a91-2131a1611c74.jpg)
![2015-10-15_144119](https://cloud.githubusercontent.com/assets/15136852/10506509/42f4cab2-734b-11e5-9e42-264a4ce493c0.jpg)

in the Petitboot menu.

**Comment by daniceexi** [comment]
It's hard for xCAT to figure out a way to handle the bootloader issue. We
will try to contact PowerKVM team to get help.

Could you manually remove the old partition before installing PowerKVM?

Thanks
Best Regards
----------------------------------------------------------------------
 Wang Xiaopeng (王晓朋)
 IBM China System Technology Laboratory
 Tel: 86-10-82453455
 Email: wxp@cn.ibm.com
 Address: 28,ZhongGuanCun Software Park,No.8 Dong Bei Wang West Road,
Haidian District Beijing P.R.China 100193

**Comment by immarvin** [comment]
I think it's reasonable to make disk detect mechanism in kickstart %pre section of powerkvm installation to be consistent with redhat7.2, ubuntu....

But there are some bugs in powerkvm kickstart parsing, opened 2 bugs to LTC:
https://bugzilla.linux.ibm.com/show_bug.cgi?id=132858
https://bugzilla.linux.ibm.com/show_bug.cgi?id=132857
````
some updates from LTC:
Hi,
This is fixed an will be out in the next build.

Please alter your your kickstart file with the proper syntax, as I mentioned in
https://bugzilla.linux.ibm.com/show_bug.cgi?id=132857#c1
````
````
Build 40 is going to be released next Wednesday (11/11).
````

**Comment by immarvin** [comment]
https://github.com/xcat2/xcat-core/pull/456

**Comment by immarvin** [comment]
https://github.com/xcat2/xcat-core/pull/531

**Comment by immarvin** [comment]
The fix to this problem is to make the code logic of determining the installation disk to be consistent across different Linux distributions.
The fix has been checked into xCAT 2.12 main branch.
The fix dependes on the fact that kickstart of PowerKVM supports %pre.  However, this does not work until PowerKVM3.1.0 release. So the fix cannot be simply back ported to xCAT 2.9.3.
I think this should as a known issue and suggest users to upgrade xCAT and use PowerKVM 3.1.0 release

**Comment by lxguang** [comment]
I have verified with xcat 2.11 and pkvm3.1. the bug can not be reproduced. Close.
